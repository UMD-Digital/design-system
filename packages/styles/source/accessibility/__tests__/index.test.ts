import * as accessibility from '../index';

describe('accessibility module', () => {
  it('should match snapshot', () => {
    expect(accessibility).toMatchSnapshot();
  });

  it('should export screenReader module', () => {
    expect(accessibility.screenReader).toBeDefined();
    expect(accessibility.screenReader.only).toBeDefined();
  });

  it('should export skip module', () => {
    expect(accessibility.skip).toBeDefined();
    expect(accessibility.skip.content).toBeDefined();
  });

  it('should have the correct nested structure', () => {
    // Test structure of exported objects
    expect(typeof accessibility.screenReader).toBe('object');
    expect(typeof accessibility.skip).toBe('object');
    
    // Test that the exported functions are correctly defined
    expect(accessibility.screenReader.only.className).toBe('sr-only');
    expect(accessibility.skip.content.className).toBe('umd-skip-content');
  });
});