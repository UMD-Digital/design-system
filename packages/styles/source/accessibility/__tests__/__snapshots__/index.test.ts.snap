// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessibility module should match snapshot 1`] = `
{
  "screenReader": {
    "only": {
      "borderWidth": "0px",
      "className": "sr-only",
      "clip": "rect(0,0,0,0)",
      "height": "1px",
      "margin": "-1px",
      "overflow": "hidden",
      "padding": "0",
      "position": "absolute",
      "whiteSpace": "nowrap",
      "width": "1px",
    },
  },
  "skip": {
    "content": {
      "&:focus": {
        "display": "block",
        "height": "inherit",
        "opacity": "1",
        "transition": "opacity 0.5s",
        "width": "auto",
        "zIndex": "9999",
      },
      "backgroundColor": "#fff",
      "className": "umd-skip-content",
      "color": "#e21833",
      "display": "block",
      "height": "0",
      "opacity": "0",
      "overflow": "hidden",
      "padding": "8px",
      "position": "absolute",
      "textDecoration": "underline",
      "transition": "none",
      "whiteSpace": "nowrap",
      "width": "0",
    },
  },
}
`;
